src/main/client/src/dashBoard/bookList/reducer.js

```diff
@@ -127,6 +150,80 @@ export const checkBook = bookId => dispatch => {
     });
 };
 
+export const checkAllBooks = user => (dispatch, getState) => {
+  let role, reservationId, borrowId, popupText;
+  const newBooks = getState().bookList.books.map(book => {
+    borrowId = null;
+    reservationId = null;
+    role = "Nothing";
+    popupText = "Click to check availability";
+    if (user.reservations) {
+      user.reservations.forEach(reservation => {
+        if (reservation.bookId === book.id) {
+          borrowId = null;
+          reservationId = reservation.id;
+          role = "Reserver";
+          popupText = "Cancel your reservation";
+        }
+      });
+    }
+    if (user.borrows) {
+      user.borrows.forEach(borrow => {
+        if (borrow.bookId === book.id && borrow.active) {
+          borrowId = borrow.id;
+          reservationId = null;
+          role = "Borrower";
+          popupText = "Return your Book";
+        }
+      });
+    }
```

Comment:
not sure the if statement is necessary? As long as user.borrows is initialised as an empty array
same for user.reservations